.github/workflows/build_and_test.yml

```diff
@@ -15,25 +15,36 @@ jobs:
         java: [8, 11.0.x, 13 ]
 
     steps:
-    - uses: actions/checkout@v2
+    - name: Checkout the code
+      uses: actions/checkout@v2
       
-    - uses: actions/cache@v1
+    - name: Cache the maven artifacts
+      uses: actions/cache@v1
       with:
         path: ~/.m2/repository
         key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
         restore-keys: |
           ${{ runner.os }}-maven-
           
-    - uses: actions/setup-java@v1
+    - name: Configure java
+      uses: actions/setup-java@v1
       with:
         java-version: ${{ matrix.java }}
 
-    - name: Build with Maven
-      run: mvn -B package --file pom.xml
+    - name: Build and run tests
+      run: mvn -B package 
+
+    - name: Run integration tests
```

Comment:
How does this decide which environment to run the integration tests against? Or is it always production

Reply:
i hadnt changed that part, so yes, i think it's always against production.

it looks like for node sdk you run it against production if on a tag, stage if on develop. i think we can add this in a separate PR / issue, just to keep the scope of this effort limited

Reply:
Sounds reasonable. I just want to make sure that when I commit a new feature to develop it doesn't fail because we haven't released that feature to the api yet